src/jit/gentree.cpp

```diff
@@ -17893,11 +17893,11 @@ bool GenTreeHWIntrinsic::OperIsMemoryLoad()
             // Avx2.BroadcastScalarToVector128/256 have vector and pointer overloads both, e.g.,
             // Vector128<byte> BroadcastScalarToVector128(Vector128<byte> value)
             // Vector128<byte> BroadcastScalarToVector128(byte* source)
-            // So, we need to check the argument's type is memory-reference (TYP_I_IMPL) or not
+            // So, we need to check the argument's type is memory-reference or not
             assert(HWIntrinsicInfo::lookupNumArgs(this) == 1);
             return (gtHWIntrinsicId == NI_AVX2_BroadcastScalarToVector128 ||
                     gtHWIntrinsicId == NI_AVX2_BroadcastScalarToVector256) &&
-                   gtOp.gtOp1->TypeGet() == TYP_I_IMPL;
+                   (gtOp.gtOp1->TypeGet() == TYP_I_IMPL || gtOp.gtOp1->TypeGet() == TYP_BYREF);
```

Comment:
Is there a general check we can do on op1 instead? To determine if it is some kind of indirection? Maybe `op1->isMemoryOp()`? If not, would checking if this is not a TYP_SIMD16 be potentially better?

@mikedn, @CarolEidt?

Reply:
The only thing that makes something an address is the fact that it is used by some kind of indirection as such. And here we're actually trying to determine is something is an indirection based on the fact that one of the operands is an address...

I suppose checking for `TYP_SIMD16` is better in the sense that it's less code. It would also handle the exotic case of `TYP_REF`, though that shouldn't be necessary.

Reply:
Thanks, will change and add comments.